dist/src/main/java/io/camunda/application/commons/console/ping/PingConsoleTask.java

```diff
@@ -0,0 +1,146 @@
+
+    } catch (final Exception e) {
+      if (e instanceof JsonProcessingException) {
+        LOGGER.warn("Failed to parse payload for Console ping task: {}", e.getMessage());
+      } else {
+        LOGGER.warn(
+            "Failed to execute Console ping task. Exception Message: {}", e.getMessage(), e);
+      }
+    }
+  }
+
+  @VisibleForTesting
+  protected void tryPingConsole(final HttpRequest request) throws RetriableException {
+    final HttpResponse<String> resp;
+    try {
+      resp = client.send(request, BodyHandlers.ofString());
+    } catch (final IOException | InterruptedException e) {
+      // If the request fails due to a network issue, we should retry.
+      throw new RetriableException("Network error: " + e.getMessage());
+    }
+
+    // We should retry on server errors, timeouts or too many request.
+    if (resp.statusCode() >= 500) {
+      throw new RetriableException("Server error: " + resp.statusCode(), resp.body());
+    } else if (resp.statusCode() == 429 || resp.statusCode() == 408) {
+      throw new RetriableException(
+          "Too many requests or timeout: " + resp.statusCode(), resp.body());
+    } else if (resp.statusCode() >= 400) {
+      // Should not retry for the remaining 4xx errors, but we log them.
+      LOGGER.warn(
+          "Received client error response: {}. No retry will be attempted.", resp.statusCode());
+    }
+  }
+
+  private String getLicensePayload() throws JsonProcessingException {
+    final ObjectMapper objectMapper = new ObjectMapper();
+    final LicensePayload.License license =
+        new LicensePayload.License(
+            managementServices.isCamundaLicenseValid(),
+            managementServices.getCamundaLicenseType().toString(),
+            managementServices.isCommercialCamundaLicense(),
+            managementServices.getCamundaLicenseExpiresAt() == null
+                ? null
+                : DATE_TIME_FORMATTER.format(managementServices.getCamundaLicenseExpiresAt()));
+    final LicensePayload payload =
+        new LicensePayload(
+            license,
+            pingConfiguration.clusterId(),
+            pingConfiguration.clusterName(),
+            pingConfiguration.properties());
+    return objectMapper.writeValueAsString(payload);
+  }
+
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  public record LicensePayload(
+      License license, String clusterId, String clusterName, Map<String, String> properties) {
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public record License(
+        boolean validLicense, String licenseType, boolean isCommercial, String expiresAt) {}
+  }
+
+  @VisibleForTesting
+  static class RetriableException extends RuntimeException {
+    public RetriableException(final String message) {
+      super(message);
+    }
+
+    public RetriableException(final String message, final String responseBody) {
+      super(message + ". Response body: " + responseBody);
```

Comment:
🙃 I'd truncate the `responseBody` for safety, you don't want an exception message with 1MiB of html in it because you misconfigure the url.